src/Hangfire.Mongo/Migration/Strategies/MongoMigrationStrategyBase.cs

```diff
@@ -41,11 +114,185 @@ protected IEnumerable<string> ExistingDatabaseCollectionNames()
         /// <summary>
         /// Generate the name of tha tbackup collection based on the original collection name and schema.
         /// </summary>
-        protected string BackupCollectionName(string collectionName, MongoSchema schema)
+        protected string GetBackupDatabaseName(string databaseName, MongoSchema schema)
+        {
+            return $@"{databaseName}-{(int)schema}-{_storageOptions.MigrationOptions.BackupPostfix}";
+        }
+
+
+        /// <summary>
+        /// Generate the name of tha tbackup collection based on the original collection name and schema.
+        /// </summary>
+        protected string GetBackupCollectionName(string collectionName, MongoSchema schema)
         {
             return $@"{collectionName}.{(int)schema}.{_storageOptions.MigrationOptions.BackupPostfix}";
         }
 
 
+        /// <summary>
+        /// Backups the collection in database identified by collectionName.
+        /// </summary>
+        /// <param name="database">Referance to the mongo database.</param>
+        /// <param name="collectionName">The name of the collection to backup.</param>
+        /// <param name="backupCollectionName">Tha name of the backup collection.</param>
+        protected virtual void BackupCollection(IMongoDatabase database, string collectionName, string backupCollectionName)
+        {
+            var dbSource = database.GetCollection<BsonDocument>(collectionName);
+            var indexes = dbSource.Indexes.List().ToList().Where(idx => idx["name"] != "_id_").ToList();
+            if (indexes.Any())
+            {
+                var dbBackup = database.GetCollection<BsonDocument>(backupCollectionName);
+                foreach (var index in indexes)
+                {
+                    var newIndex = new BsonDocumentIndexKeysDefinition<BsonDocument>(index["key"].AsBsonDocument);
+                    var newIndexKeys = index.Names.ToList();
+
+                    var newOptions = new CreateIndexOptions();
+                    foreach (var mapping in CreateIndexOptionsMapping.Where(m => newIndexKeys.Contains(m.Key)))
+                    {
+                        var prop = newOptions.GetType().GetTypeInfo().GetProperty(mapping.Name);
+                        if (prop != null)
+                        {
+                            prop.SetValue(newOptions, mapping.Convert(index[mapping.Key].Value));
+                        }
+                    }
+                    dbBackup.Indexes.CreateOne(newIndex, newOptions);
+                }
+            }
+
+            var aggregate = new BsonDocument(new Dictionary<string, object>
+            {
+                {
+                    "aggregate", collectionName
```

Comment:
Consider using the C# 6 Dictionary initilizer. eg.
["aggregate"] = collectionName

Reply:
Yes, tnx. I do prefer the old notation as it resembles json more - which in this particular place is better IMHO